extras/retry_test.go

```diff
@@ -0,0 +1,363 @@
+package extras
+
+import (
+	"log"
+	"net/http"
+	"os"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/orian/flakyhttp"
+	posthog "github.com/posthog/posthog-go"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+// retryTestCallback implements posthog.Callback to track success/failure
+type retryTestCallback struct {
+	t            *testing.T
+	mu           sync.Mutex
+	successCount int
+	failureCount int
+	successChan  chan posthog.APIMessage
+	failureChan  chan error
+}
+
+func newRetryTestCallback(t *testing.T) *retryTestCallback {
+	return &retryTestCallback{
+		t:           t,
+		successChan: make(chan posthog.APIMessage, 100),
+		failureChan: make(chan error, 100),
+	}
+}
+
+func (c *retryTestCallback) Success(msg posthog.APIMessage) {
+	c.mu.Lock()
+	c.successCount++
+	c.mu.Unlock()
+	c.t.Logf("Callback: SUCCESS for event")
+	c.successChan <- msg
+}
+
+func (c *retryTestCallback) Failure(msg posthog.APIMessage, err error) {
+	c.mu.Lock()
+	c.failureCount++
+	c.mu.Unlock()
+	c.t.Logf("Callback: FAILURE - %v", err)
+	c.failureChan <- err
+}
+
+func (c *retryTestCallback) GetCounts() (success, failure int) {
+	c.mu.Lock()
+	defer c.mu.Unlock()
+	return c.successCount, c.failureCount
+}
+
+// TestRetryOnEOF_EventualSuccess tests that PostHog retries on EOF and eventually succeeds.
+func TestRetryOnEOF_EventualSuccess(t *testing.T) {
+	// Server fails 3 times with EOF, then succeeds
```

Comment:
The comment states that we expect 3 failures and 1 success at the end, but there's no verification of this. Would it make sense to assert that `server.RequestCount()` is exactly 4?